test/dist_model_parallel_test.py

```diff
@@ -0,0 +1,138 @@
+import torch
+import torch.nn as nn
+import torch.distributed.autograd as dist_autograd
+import torch.distributed.rpc as rpc
+# from torch.distributed.optim import DistributedOptimizer
+# from torch.distributed.rpc import RRef
+
+from dist_utils import INIT_METHOD_TEMPLATE, dist_init, TEST_CONFIG
+
+import unittest
+
+def _call_method(method, obj_rref, *args, **kwargs):
+    return method(obj_rref.local_value(), *args, **kwargs)
+
+
+def _remote_method(method, obj_rref, *args, **kwargs):
+    return rpc.rpc_sync(
+        obj_rref.owner(),
+        _call_method,
+        args=[method, obj_rref] + list(args),
+        kwargs=kwargs
+    )
+
+class RemoteModule(nn.Module):
```

Comment:
What is the reason for RemoteModule in here? I believe RemoteModule only makes sense as a wrapper module that wraps regular modules. The way it's exposed here you'll still have to implement a special module that inherits from RemoteModule, so there's not much utility for this class.

If the only functionality is exposing remote_parameters(), then simply make it a free function, e.g.:

def get_module_param_rrefs(nn_module: NNModule):
    return [RRef(param) for param in self.parameters()]

If the goal is to discover remote modules, you could have a RRef wrapper on the client side instead (not on the remote side).